src/lib.rs

```diff
@@ -1595,6 +1734,24 @@ impl<'a> MatchTreeStep<'a> {
                     Self::from_repeat_count(module, bounds.min, a, Rc::new(Next::Empty))
                 }
             }
+            Format::RepeatBetween(xmin, xmax, a) => {
+                let min_bounds = xmin.bounds();
+                let max_bounds = xmax.bounds();
+                match (min_bounds.is_exact(), max_bounds.is_exact()) {
+                    (Some(min), Some(max)) => match min.cmp(&max) {
+                        Ordering::Less => {
+                            Self::from_repeat_between(module, (min, max), a, next.clone())
+                        }
+                        Ordering::Equal => Self::from_repeat_count(module, min, a, next.clone()),
+                        Ordering::Greater => {
+                            panic!("incoherent repeat-between: min {} > max {}", min, max)
+                        }
+                    },
+                    _ => {
+                        unreachable!("inexact repeat-between bounds (not technically a problem but not what the combinator was designed for...");
```

Comment:
Similarly here.

Reply:
Same as above, it is hard to figure out without an Expr-evaluator what number of repeats are 'acceptable' in a dynamic context. Say, for example:

```rust
record([
    ("min", base.u32be()),
    ("delta", base.u32be()),
    ("seq", repeat_between(var("min"), Expr::Arith(Arith::Add, Box::new(var("min")), Box::new(var("delta"))), base.u8())
])
```

How would one construct a match-tree for this?